Consumer search criteria not cleared/used when navigating to another …

jithishkumarg opened the pull request:
…consumer - fixed

Fixes #1221 

The search text will be cleared when switched to different cluster's consumers

<!-- ignore-task-list-start -->
- [x] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)
Not a breaking change - issue fix just focuses on the search filter

<!-- ignore-task-list-end -->
**What changes did you make?** (Give an overview)
When switching between different clusters, I emptied the search field if the query param is not available

**Is there anything you'd like reviewers to focus on?**
No, please let me know if anything needs to be taken care of as this is my first open source contribution

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)
<!-- ignore-task-list-start -->
- [x] No need to
- [x ] Manually (please, describe, if necessary) - Manually
- [x] Unit checks
- [x] Integration checks
- [x] Covered by existing automation
<!-- ignore-task-list-end -->

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [ x] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x ] My changes generate no new warnings (e.g. Sonar is happy)
- [ x] I have added tests that prove my fix is effective or that my feature works
- [x ] New and existing unit tests pass locally with my changes
- [ x] Any dependent changes have been merged

Check out [Contributing](https://github.com/kafbat/kafka-ui/blob/main/.github/CONTRIBUTING.md) and [Code of Conduct](https://github.com/kafbat/kafka-ui/blob/main/.github/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * Improved search input synchronization with URL query and component props to prevent unexpected input resets, avoid unnecessary DOM updates, and handle controlled vs. URL-driven states more reliably.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jithishkumarg committed:
Consumer search criteria not cleared/used when navigating to another consumer - fixed
The search text will be cleared when switched to different cluster's consumers

HurSungYun commented:
@jithishkumarg I'm experiencing the same issue https://github.com/kafbat/kafka-ui/issues/1588, so I hope this PR to be merged. Can you please look at the failing tests when you have time?

jithishkumarg committed:
fixed the cause for the failing test cases
fix(search): sync from q param only when present and preserve controlled input behavior

jithishkumarg committed:
fixes - indentation issue

jithishkumarg committed:
addressing issues highlighted by the coderabbitai tool

jithishkumarg commented on frontend/src/components/common/Search/Search.tsx:
```diff
@@ -32,10 +32,14 @@ const Search: React.FC<SearchProps> = ({
+      ref.current.value = qParam;
     }
-  }, [value]);
+  }, [searchParams, onChange, value]);
```
added the fix for the issue thus resolving this conversation.

jithishkumarg replied:
this is fixed.

jithishkumarg commented:
Hi @Haarolean I have addressed the issue highlighted by coderabbitai tool. Thanks

germanosin committed:
Merge branch 'main' into issues/1221

jithishkumarg committed:
fixing failed test cases

jithishkumarg committed:
indentation

jithishkumarg committed:
Merge branch 'issues/1221' of https://github.com/jithishkumarg/jithishkumarg-kafka-ui into issues/1221

jithishkumarg commented:
Hi @Haarolean what is the next step on this? Can you guide me?
